nilearn/decoding/searchlight.py

```diff
@@ -353,7 +382,58 @@ def fit(self, imgs, y, groups=None):
             self.n_jobs,
             self.verbose,
         )
+        self.process_mask_ = process_mask
+        self.masked_scores_ = scores
         scores_3D = np.zeros(process_mask.shape)
-        scores_3D[process_mask] = scores
+        scores_3D[np.where(process_mask)] = scores
```

Comment:
Why do you introduce the np.where` here ?

Reply:
extracts the coordinates of non-zero elements

Reply:
You don't want to do that, but rather use fancy indexing instead.